[build-tooling-libs] Add option to specify the swiftc compiler to build compiler sources

ahoppen opened the pull request:
This is needed to build the most recent state of main if it uses C++ interop features that aren't in the host's toolchain yet.

ahoppen commented:
@swift-ci Please smoke test

rintaro commented on utils/build-tooling-libs:
```diff
@@ -172,10 +173,14 @@ class Builder(object):
             "-DSWIFT_HOST_VARIANT_ARCH=" + self.arch,
             "-DCMAKE_Swift_COMPILER_TARGET=" + host_triple,
-            "-DCMAKE_Swift_COMPILER=" + self.toolchain.swiftc,
+            "-DCMAKE_Swift_COMPILER=" + self.swiftc,
             "-DCMAKE_C_FLAGS=" + llvm_c_flags,
             "-DCMAKE_CXX_FLAGS=" + llvm_c_flags,
         ]
+        if self.swiftc:
+            cmake_args += [
+                f"-DCMAKE_Swift_COMPILER={self.swiftc}"
+            ]
```
Hmm I think this is not correct. I think you meant default to use `self.toolchain.swiftc` but override it with `self.swiftc`. But the actual code doesn't use `self.toolchain.swiftc`?

ahoppen replied:
Oh, yes. That’s an artifact from an old version that I should have deleted.

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please smoke test

rintaro commented on utils/build-tooling-libs:
```diff
@@ -499,6 +501,16 @@ Example invocations:
         help="the path to ninja (default = %s)" % default_ninja,
     )
+    option(
+        "--swiftc",
```
Not a strong opinion but `--swiftc-path` would be more consistent with `--ninja-path`

ahoppen replied:
You’re right 👍 Thanks.

ahoppen committed:
[build-tooling-libs] Add option to specify the swiftc compiler to build compiler sources

This is needed to build the most recent state of main if it uses C++ interop features that aren't in the host's toolchain yet.

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please smoke test macOS
